### ISSUE TITLE
Move flaw details in PR reviewer to GitHub Annotations

### ISSUE THREAD

**Issue by nschonni** [mdn/fred#955]
I think the preview URLs are best suited as a comment on the PR, but it might be better to use the Checks/Annotations/Problem Matcher setup for the flaws portion of the PR check.
I know getting back to the specific line numbers for some of the flaw results might not be possible, but I think setting the "line" part of the match to 1, would just add the comment on the first line of the file.

Some related discussions on https://github.com/mdn/yari/pull/2753

**Comment by peterbe** [comment]
@ddbeck Was this your suggestion too?

**Comment by peterbe** [comment]
In principle, I like that idea. We use PyGitHub which is a very capable lib for posting comments. So it probably would be quite possible. But it's a bit less trivial. Worth a shot maybe. 

Also, if a flaw does not have a `line` we could simply say that in the comment which might explain the confusion when someone looks at line 1 and wonders "What's that got to do with this line??"

**Comment by nschonni** [comment]
> In principle, I like that idea. We use PyGitHub which is a very capable lib for posting comments. So it probably would be quite possible. But it's a bit less trivial. Worth a shot maybe.

If you add and register a problem matcher https://github.com/mdn/content/pull/1206/files#diff-25c661e20a38ae8e4610534138256b0595a4703ee1a06430bc22add458e56b8dR1 you don't even need the PyGitHub part at all 😄 . It creates those annotations based on the format of the messages the deployment would print to the log stream

**Comment by peterbe** [comment]
Oh I see. I guess I could have guessed it was a specific thing when you wrote "GitHub Annotations" as a noun. So it's not simply PR comments for specific lines on specific lines. 

If you want to experiment more deeply with this, we can use my fork of mdn/content and we can set up some PRs within that fork which we can bombard to test things. I'd love to learn more about how this works because right now I know next to nothing.

**Comment by ddbeck** [comment]
> @ddbeck Was this your suggestion too?

Yes, pretty much. I was thinking of setting the checks run `output.summary` and `output.text` ([_the properties of the `output` object_ in the GitHub docs](https://docs.github.com/en/rest/reference/checks#output-object)) to the Markdown that currently goes into the bot comment, but using annotations would be a nice upgrade (especially for stuff that's per-file). I think @nschonni has it right that a problem matcher would be easier to implement, too.

**Comment by caugner** [comment]
This would be still nice to have.

**Comment by Guilherme356** [comment]
https://github.com/orgs/mdn/projects/25/views/1/* Set font size on all <p> elements */
p {
  font-size: 12px;
  color: rebeccapurple;
}
/* Set font size on all <p> elements */
p {
  font-size: 12px;
  color: rebeccapurple;
}

**Comment by caugner** [comment]
As part of adding a test workflow to mdn/curriculum, I looked into surfacing flaws as GitHub Annotations, see [this workflow draft](https://gist.github.com/caugner/be1254f9e837ed8c898cc84e28be40b3).

Unfortunately, the `issues.json` generated by Rari currently returns 0-based line numbers where line 0 is the first line after the frontmatter. We would either have to recalculate the line numbers when creating the annotations, or have Rari return 1-based line numbers where line 0 is the first line of the file. This currently blocks us from moving forward.

Notes:

- The new "Files changed" UI no longer shows GitHub Annotations in unchanged files, which was a nice feature of the old UI.
- The GitHub Checks API has some limitations:
  > The Checks API limits the number of annotations to a maximum of 50 per API request. To create more than 50 annotations, you have to make multiple requests to the [Update a check run](https://docs.github.com/rest/checks/runs#update-a-check-run) endpoint. Each time you update the check run, annotations are appended to the list of annotations that already exist for the check run. GitHub Actions are limited to 10 warning annotations and 10 error annotations per step.

**Comment by caugner** [comment]
Moving to backlog.
